app/(main)/o/[slug]/header.tsx

```diff
@@ -51,17 +53,34 @@ const HeaderPopoverContent = ({
   </PopoverContent>
 );
 
-export default function Header({ isAnonymous, tenant, name, email, onNavClick = () => {} }: Props) {
+export default function Header({ isAnonymous, tenant, name, email, onNavClick = () => {}, hasSession }: Props) {
   const router = useRouter();
   const [tenants, setTenants] = useState<z.infer<typeof tenantListResponseSchema>>([]);
+  const [conversationId, setConversationId] = useState<string>("");
+  const pathname = usePathname();
 
   useEffect(() => {
-    (async () => {
-      const res = await fetch("/api/tenants");
-      const tenants = tenantListResponseSchema.parse(await res.json());
-      setTenants(tenants);
-    })();
-  }, []);
+    // Only fetch tenants if there is an active session
+    if (hasSession) {
+      (async () => {
+        const res = await fetch("/api/tenants");
+        if (res.ok) {
+          const tenants = tenantListResponseSchema.parse(await res.json());
+          setTenants(tenants);
+        }
+      })();
+    }
+  }, [hasSession]);
```

Comment:
We should probably start making some custom hooks.  This component is getting a little messy.  Maybe something like `useFetchTenants`?  I also think we might want to look at `better-fetch` as a way to make JSON api calls a little cleaner.